packages/js/settings-editor/typings/global.d.ts

```diff
@@ -0,0 +1,53 @@
+type SettingField = {
+    title?: string;
+    type: string;
+    id: string;
+    desc?: string;
+    desc_tip?: boolean | string;
+    default?: string | number | boolean | object;
+    value: string | number | boolean | object;
+    placeholder?: string;
+    custom_attributes?: {
+        [key: string]: string | number;
+    };
+    options?: {
+		[key: string]: string;
+	};
+    css?: string;
+    class?: string;
+    checkboxgroup?: 'start' | 'end' | '';
+    autoload?: boolean;
+    show_if_checked?: string;
+    content?: string;
+    [key: string]: any;
+};
+
+type SettingsSection = {
+    label: string;
+    settings: SettingField[];
+};
+
+type SettingsPageData = {
+    label: string;
+    sections: {
+        [key: string]: SettingsSection;
+    };
+    is_modern: boolean;
+};
+
+type SettingsData = {
+    [key: string]: SettingsPageData;
+};
+
+declare global {
+	interface Window {
+		wcSettings?: {
+			admin?: {
+				settingsPages: SettingsData;
+			};
+		};
+	}
+}
+
```

Comment:
So having these in `global.d.ts` allows us to not have to import these types?

Reply:
It allows us to define types for the global `window.wcSettings` object. This way, we don't need to import types for each file that uses `window.wcSettings`. We can use typeof `window.wcSettings` to get the types.